aioesphomeapi/client.py

```diff
@@ -518,34 +518,46 @@ def unsub() -> None:
                 self._connection.remove_message_callback(on_msg)
 
         try:
```

Comment:
Bleak does something similar here https://github.com/hbldh/bleak/blob/master/bleak/backends/bluezdbus/client.py#L159 but catches `BaseException`

I think catching `asyncio.CancelledError` is probably enough here though.

I wasn't able to get it to leak anymore.